### PR TITLE
Circuit for opcode TIMESTAMP

### PR DESCRIPTION
**Pull request by z2trillion**
Implement the TimestampGadget for opcode TIMESTAMP.

Specs: https://github.com/scroll-tech/zkevm-specs/pull/31

Other changes:
- Use `StackOnlyOpcode::<0>` for COINBASE bus mapping.
- Change mock block author address to a non-default value, so that it doesn't match the value in a default `BlockContext`.

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/witness.rs [hunk 1]
```diff
@@ -63,7 +63,10 @@ impl<F: FieldExt> BlockContext<F> {
                 [
                     F::from(BlockContextFieldTag::Time as u64),
                     F::zero(),
-                    F::from(self.time),
+                    RandomLinearCombination::random_linear_combine(
+                        self.time.to_le_bytes(),
+                        randomness,
+                    ),
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/witness.rs:69]: why this change? the field is big enough to contain u64?

File: bus-mapping/src/evm/opcodes/timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#[cfg(test)]
+mod coinbase_tests {
```
Comment (icemelon) [inline bus-mapping/src/evm/opcodes/timestamp.rs:2]:
```suggestion
mod timesamp_tests {
```

File: zkevm-circuits/src/evm_circuit/witness.rs (hunk 1 above)
Comment (z2trillion) [reply zkevm-circuits/src/evm_circuit/witness.rs]: thanks for pointing this out. I updated the gadget so that this change isn't needed anymore.

**Review by lispc — APPROVED** [review]

**Review by icemelon — COMMENTED** [review]
Need to rebase. Otherwise, LGTM.

Will wait for spec PR ready before sending PR to upstream

**Review by icemelon — APPROVED** [review]

**Comment by icemelon** [comment]
Close and move to https://github.com/appliedzkp/zkevm-circuits/pull/307
